### PR TITLE
feat(tsconfig): expose "baseUrl" and "paths" in compilerOptions

### PR DESCRIPTION
**Pull request by hassanazharkhan**
Added missing props in `TypeScriptCompilerOptions` which are needed to re-map the imports.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .all-contributorsrc [hunk 1]
```diff
@@ -477,6 +477,69 @@
       "contributions": [
         "code"
       ]
+    },
```
Comment (Chriscbr) [inline .all-contributorsrc:480]: Can you revert the changes to this file and the README? We should keep this PR focused on one change and update the contributors list in a separate PR.

File: src/typescript-config.ts [hunk 2]
```diff
@@ -345,6 +345,22 @@ export interface TypeScriptCompilerOptions {
...
+  readonly baseUrl?: string;
+
+  /**
+   * A series of entries which re-map imports to lookup locations relative to the baseUrl, there is a larger coverage of paths in the handbook.
+   * paths lets you declare how TypeScript should resolve an import in your require/imports.
+   *
+   * @see https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping
+   * @default undefined
+  */
+  readonly paths?: { [key: string]: string[] };
```
Comment (Chriscbr) [inline src/typescript-config.ts:363]: There should also be a change to API.md - it should be automatically updated when you run the compile task.

**Review by Chriscbr — CHANGES_REQUESTED** [review]

File: .all-contributorsrc (hunk 1 above)
Comment (hassanazharkhan) [reply .all-contributorsrc]: Done

File: src/typescript-config.ts (hunk 2 above)
Comment (hassanazharkhan) [reply src/typescript-config.ts]: Done

**Review by Chriscbr — APPROVED** [review]

File: src/__tests__/python/__snapshots__/poetry.test.ts.snap [hunk 3]
```diff
@@ -147,6 +147,11 @@ cython_debug/
         "type": "runtime",
         "version": "0.0.2",
       },
+      Object {
```
Comment (Chriscbr) [inline src/__tests__/python/__snapshots__/poetry.test.ts.snap:150]: I think if you just omit this file from the PR, the build should pass.

Comment (hassanazharkhan) [reply src/__tests__/python/__snapshots__/poetry.test.ts.snap]: Done
